src/single/index.js

```diff
@@ -3,15 +3,12 @@
     var loadingTimer = 0;
 
     var infrastructure = {
-        addQueryStringToLink: function(queryList) {
-            if (!queryList || !Array.isArray(queryList) || queryList.length === 0) return false;
+        getLink: function(queryList, link) {
+          if (!queryList || !Array.isArray(queryList) || queryList.length === 0) return false;
 
```

Comment:
even if the list were empty, the function should return link instead a value of false